vendor/github.com/containers/image/v5/docker/docker_image_src.go

```diff
@@ -121,7 +121,7 @@ func newImageSource(ctx context.Context, sys *types.SystemContext, ref dockerRef
 // The caller must call .Close() on the returned ImageSource.
 func newImageSourceAttempt(ctx context.Context, sys *types.SystemContext, logicalRef dockerReference, pullSource sysregistriesv2.PullSource,
 	registryConfig *registryConfiguration) (*dockerImageSource, error) {
-	physicalRef, err := newReference(pullSource.Reference)
+	physicalRef, err := newReference(pullSource.Reference, logicalRef.isUnknownDigest)
```

Comment:
I would expect the code to completely reject unknown-digest references somewhere around the start of `newImageSource`; I don’t know what “pull from a digest which is unknown” could mean.